tee-worker/litentry/core/credentials/src/nodereal/crypto_summary/mod.rs

```diff
@@ -0,0 +1,429 @@
+		let mut flag_eth_nft: [bool; 15] = [false; 15];
+
+		// ETH and BNB holder use different APIs than other tokens, so they need to be handled separately
+		let mut is_eth_holder = false;
+		let mut is_bnb_holder = false;
+
+		for (network, addresses) in identities {
+			if *network == Web3Network::Bsc {
+				// Query Token
+				for address in addresses {
+					let res = self
+						.bsc_client
+						.get_token_holdings(address)
+						.map_err(|e| e.into_error_detail())?;
+
+					let result: ResponseTokenResult =
+						serde_json::from_value(res.result).map_err(|_| ErrorDetail::ParseError)?;
+					let mut token_addresses = vec![];
+					if let Some(details) = result.details {
+						details.iter().for_each(|detail| {
+							token_addresses.push(detail.token_address.clone());
+						});
+						Self::update_holding_flag(
+							&mut flag_bsc_token,
+							&CRYPTO_SUMMARY_TOKEN_ADDRESSES_BSC,
+							&token_addresses,
+						);
+					}
+
+					// Query BNB balance on BSC
+					if !is_bnb_holder {
+						let native_balance = result.native_token_balance;
+						let balance = get_native_token_balance(&native_balance);
+						if balance > 0.0 {
+							is_bnb_holder = true;
+						}
+					}
+				}
+			}
+
+			if *network == Web3Network::Ethereum {
+				for address in addresses {
+					// Query Token
+					let res_token = self
+						.eth_client
+						.get_token_holdings(address)
+						.map_err(|e| e.into_error_detail())?;
+					let result: ResponseTokenResult = serde_json::from_value(res_token.result)
+						.map_err(|_| ErrorDetail::ParseError)?;
+
+					let mut token_addresses = vec![];
+					if let Some(details) = result.details {
+						details.iter().for_each(|detail| {
+							token_addresses.push(detail.token_address.clone());
+						});
+						Self::update_holding_flag(
+							&mut flag_eth_token,
+							&CRYPTO_SUMMARY_TOKEN_ADDRESSES_ETH,
+							&token_addresses,
+						);
+					}
+
+					// Query ETH balance on Ethereum
+					if !is_eth_holder {
+						let native_balance = result.native_token_balance;
+						let balance = get_native_token_balance(&native_balance);
+						if balance > 0.0 {
+							is_eth_holder = true;
+						}
+					}
+
+					// Query NFT
+					let param = GetNFTHoldingsParam {
+						account_address: address.to_string(),
+						token_type: "ERC721".to_string(),
+						page: 1,
+						page_size: 100,
+					};
+
+					let res_nft = self
+						.eth_client
+						.get_nft_holdings(&param)
+						.map_err(|e| e.into_error_detail())?;
+
+					error!("nft token res: {:?}", res_nft);
+
+					let details = res_nft.details;
+
+					let mut nft_addresses = vec![];
+					details.iter().for_each(|detail| {
+						nft_addresses.push(detail.token_address.clone());
+					});
+
+					Self::update_holding_flag(
+						&mut flag_eth_nft,
+						&CRYPTO_SUMMARY_NFT_ADDRESSES_ETH,
+						&nft_addresses,
+					)
+				}
+			}
+		}
+
+		// Update ETH and BNB
+		flag_bsc_token[11] = is_bnb_holder;
+		flag_eth_token[14] = is_eth_holder;
```

Comment:
I'm not a super fan of using such hardcoded index (same applies to the string tuple), but fine as long as it works for now